src/backend/distributed/replication/multi_logical_replication.c

```diff
@@ -564,7 +564,7 @@ DropAllLogicalReplicationLeftovers(LogicalRepType type)
 	 * run in a transaction. By forcing a new connection we make sure no
 	 * transaction is active on the connection.
 	 */
-	int connectionFlags = FORCE_NEW_CONNECTION;
+	int connectionFlags = OUTSIDE_TRANSACTION;
```

Comment:
I'm not 100% sure about this one. We close all these connections at the end of this function. But probably that's fine. It's probably much better to reopen the few that we need, than to keep the ones we don't open until the move is complete.

Reply:
I think it is a little unusual to miss  FORCE_NEW_CONNECTION and then have CloseConnection. Close'ing non-force-new connections are typically done in the connection_mngmnt.c, which would be nice to keep.

I'd either use `FORCE_NEW_CONNECTION` and keep CloseConnection, or remove `CloseConnection`.

Given all our efforts in this patch, I cannot see a reason not to remove `CloseConnection` here?

Reply:
I changed it back to FORCE_NEW_CONNECTION. The main reason I prefer to keep CloseConnection, is because this opens superuser connections to all nodes. Not just to the ones that we are involved in the move. Since moving a shard can take a long time, it seems preferable to close these unnecessary connections during the move.